pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,385 @@
+    strain = strain[:, [0, 4, 8, 5, 2, 1]]
+    strain[:,3:] *= 2
+    if higher_terms is not None:
+        strain = np.concatenate((strain, higher_terms), axis=-1)
+    stress = np.einsum('nik,mkl,njl->nmij',
+                       rotations, stress, rotations).reshape(-1, 9)
+    stress = stress[:, [0, 4, 8, 5, 2, 1]]
+    score = []
+    coeff = []
+    for ii in range(6):
+        strain_tmp = strain.copy()
+        strain_tmp[:, 6:] *= strain[:, ii, np.newaxis]
+        reg = LinearRegression(
+            fit_intercept=fit_first_order).fit(strain_tmp, stress[:,ii])
+        coeff.append(reg.coef_)
+        score.append(reg.score(strain_tmp, stress[:,ii]))
+    return np.array(coeff), np.array(score)
+
+def _fit_coeffs_with_energies(
+        strain,
+        energy,
+        volume,
+        rotations,
+        max_polynomial_order=None,
+        fit_first_order=False,
+    ):
+    higher_terms = _get_higher_order_terms(
+        strain,
+        max_polynomial_order=max_polynomial_order,
+        rotations=rotations,
+        derivative=False,
+    )
+    strain_voigt = np.einsum('nik,mkl,njl->nmij',
+                             rotations, strain, rotations).reshape(-1, 9)
+    strain_voigt = strain_voigt[:, [0, 4, 8, 5, 2, 1]]
+    strain_voigt[:,3:] *= 2
+    energy = np.tile(energy, len(rotations))
+    volume = np.tile(volume, len(rotations))
+    strain = 0.5*np.einsum('ni,nj->nij', strain_voigt, strain_voigt)
+    strain = np.triu(strain).reshape(-1, 36)
+    strain = strain[:,np.sum(strain, axis=0)!=0]
+    if higher_terms is not None:
+        strain = np.concatenate((strain, higher_terms), axis=-1)
+    if fit_first_order:
+        strain = np.concatenate((strain, strain_voigt), axis=-1)
+    strain = np.einsum('n,ni->ni', volume, strain)
+    reg = LinearRegression().fit(strain, energy)
+    score = reg.score(strain, energy)
+    coeff = np.triu(np.ones((6,6))).flatten()
+    coeff[coeff!=0] *= reg.coef_[:21]*eV_div_A3_to_GPa
+    coeff = coeff.reshape(6,6)
+    coeff = 0.5*(coeff+coeff.T)
+    return coeff, score
+
+def _get_linear_dependent_indices(strain_lst):
+    """
+    This function returns an array of booleans, which shows linearly dependent
+    strain matrices. Strain matrices which have True along axis=0 are linearly
+    dependent. Axis=1 should contain the total number of strain matrices.
+    """
+    s = np.array(strain_lst).reshape(-1, 9)
+    norm = np.linalg.norm(s, axis=-1)
+    indices = np.round(np.einsum('ni,n,n->ni', s, np.sign(np.sum(s, axis=-1)),
+                       1/(norm+np.isclose(norm, 0))), decimals=8)
+    indices = np.unique(indices, axis=0, return_inverse=True)[1]
+    na = np.newaxis
+    indices = np.unique(indices[~np.isclose(norm, 0)])[:,na]==indices[na,:]
+    # 0-tensor gets a special treatment, since it's linearly dependent on all
+    # terms (even though it virtually changes nothing)
+    indices[:,np.isclose(norm, 0)] = True
```

Comment:
PEP8: space --- wait, actually it might not be strictly PEP8 to put a space after a comma in the `[]`. But I think it would be more consistent with your style elsewhere.